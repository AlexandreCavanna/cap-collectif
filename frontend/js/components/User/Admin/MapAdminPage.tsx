import * as React from 'react'
import { FormattedMessage } from 'react-intl'
import { QueryRenderer, graphql } from 'react-relay'
import type { MapAdminPageQueryResponse } from '~relay/MapAdminPageQuery.graphql'
import environment, { graphqlError } from '../../../createRelayEnvironment'
import Loader from '../../Ui/FeedbacksIndicators/Loader'
import MapboxAdminConfig from './MapboxAdminConfig'

const mapboxAdminConfig = ({
  error,
  props,
}: ReactRelayReadyState & {
  props?: MapAdminPageQueryResponse | null | undefined
}) => {
  if (error) {
    console.log(error) // eslint-disable-line no-console

    return graphqlError
  }

  if (props) {
    if (props.mapToken) {
      return <MapboxAdminConfig {...props} />
    }

    return graphqlError
  }

  return <Loader />
}

const MapAdminPage = () => (
  <div className="box box-primary container-fluid">
    <div className="box-header">
      <h3 className="box-title">
        <FormattedMessage id="customize" />
      </h3>
    </div>
    <div className="box-content">
      <QueryRenderer
        environment={environment}
        query={graphql`
          query MapAdminPageQuery($provider: MapTokenProvider!) {
            mapToken(provider: $provider, includeDefault: false) {
              ...MapboxAdminConfig_mapToken
            }
          }
        `}
        variables={{
          provider: 'MAPBOX',
        }}
        render={mapboxAdminConfig}
      />
    </div>
  </div>
)

export default MapAdminPage
