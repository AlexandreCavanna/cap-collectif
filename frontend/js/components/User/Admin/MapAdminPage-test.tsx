/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import MapAdminPage from './MapAdminPage'

describe('<MapAdminPage/>', () => {
  it('should render', () => {
    const wrapper = shallow(<MapAdminPage />)
    expect(wrapper).toMatchSnapshot()
  })
})
