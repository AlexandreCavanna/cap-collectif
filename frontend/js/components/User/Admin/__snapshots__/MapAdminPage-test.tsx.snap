// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MapAdminPage/> should render 1`] = `
<div
  className="box box-primary container-fluid"
>
  <div
    className="box-header"
  >
    <h3
      className="box-title"
    >
      <FormattedMessage
        id="customize"
        values={Object {}}
      />
    </h3>
  </div>
  <div
    className="box-content"
  >
    <ReactRelayQueryRenderer
      environment={"RelayModernEnvironment()"}
      query={
        Object {
          "default": Object {
            "fragment": Object {
              "abstractKey": null,
              "argumentDefinitions": Array [
                Object {
                  "defaultValue": null,
                  "kind": "LocalArgument",
                  "name": "provider",
                },
              ],
              "kind": "Fragment",
              "metadata": null,
              "name": "MapAdminPageQuery",
              "selections": Array [
                Object {
                  "alias": null,
                  "args": Array [
                    Object {
                      "kind": "Literal",
                      "name": "includeDefault",
                      "value": false,
                    },
                    Object {
                      "kind": "Variable",
                      "name": "provider",
                      "variableName": "provider",
                    },
                  ],
                  "concreteType": "MapToken",
                  "kind": "LinkedField",
                  "name": "mapToken",
                  "plural": false,
                  "selections": Array [
                    Object {
                      "args": null,
                      "kind": "FragmentSpread",
                      "name": "MapboxAdminConfig_mapToken",
                    },
                  ],
                  "storageKey": null,
                },
              ],
              "type": "Query",
            },
            "hash": "697628a7acbf7e49cb1087504139263e",
            "kind": "Request",
            "operation": Object {
              "argumentDefinitions": Array [
                Object {
                  "defaultValue": null,
                  "kind": "LocalArgument",
                  "name": "provider",
                },
              ],
              "kind": "Operation",
              "name": "MapAdminPageQuery",
              "selections": Array [
                Object {
                  "alias": null,
                  "args": Array [
                    Object {
                      "kind": "Literal",
                      "name": "includeDefault",
                      "value": false,
                    },
                    Object {
                      "kind": "Variable",
                      "name": "provider",
                      "variableName": "provider",
                    },
                  ],
                  "concreteType": "MapToken",
                  "kind": "LinkedField",
                  "name": "mapToken",
                  "plural": false,
                  "selections": Array [
                    Object {
                      "alias": null,
                      "args": null,
                      "kind": "ScalarField",
                      "name": "id",
                      "storageKey": null,
                    },
                    Object {
                      "alias": null,
                      "args": null,
                      "kind": "ScalarField",
                      "name": "publicToken",
                      "storageKey": null,
                    },
                    Object {
                      "alias": null,
                      "args": null,
                      "kind": "ScalarField",
                      "name": "secretToken",
                      "storageKey": null,
                    },
                    Object {
                      "alias": null,
                      "args": null,
                      "kind": "ScalarField",
                      "name": "provider",
                      "storageKey": null,
                    },
                    Object {
                      "alias": null,
                      "args": null,
                      "kind": "ScalarField",
                      "name": "styleOwner",
                      "storageKey": null,
                    },
                    Object {
                      "alias": null,
                      "args": null,
                      "kind": "ScalarField",
                      "name": "styleId",
                      "storageKey": null,
                    },
                    Object {
                      "alias": null,
                      "args": null,
                      "concreteType": "MapTokenStyle",
                      "kind": "LinkedField",
                      "name": "styles",
                      "plural": true,
                      "selections": Array [
                        Object {
                          "alias": null,
                          "args": null,
                          "kind": "ScalarField",
                          "name": "id",
                          "storageKey": null,
                        },
                        Object {
                          "alias": null,
                          "args": null,
                          "kind": "ScalarField",
                          "name": "owner",
                          "storageKey": null,
                        },
                        Object {
                          "alias": null,
                          "args": null,
                          "kind": "ScalarField",
                          "name": "name",
                          "storageKey": null,
                        },
                        Object {
                          "alias": null,
                          "args": null,
                          "kind": "ScalarField",
                          "name": "previewUrl",
                          "storageKey": null,
                        },
                        Object {
                          "alias": null,
                          "args": null,
                          "kind": "ScalarField",
                          "name": "createdAt",
                          "storageKey": null,
                        },
                        Object {
                          "alias": null,
                          "args": null,
                          "kind": "ScalarField",
                          "name": "updatedAt",
                          "storageKey": null,
                        },
                        Object {
                          "alias": null,
                          "args": null,
                          "kind": "ScalarField",
                          "name": "isCurrent",
                          "storageKey": null,
                        },
                      ],
                      "storageKey": null,
                    },
                  ],
                  "storageKey": null,
                },
              ],
            },
            "params": Object {
              "cacheID": "e960c6d61e151ea0b4e57def543dbcdb",
              "id": null,
              "metadata": Object {},
              "name": "MapAdminPageQuery",
              "operationKind": "query",
              "text": "query MapAdminPageQuery(
  $provider: MapTokenProvider!
) {
  mapToken(provider: $provider, includeDefault: false) {
    ...MapboxAdminConfig_mapToken
    id
  }
}

fragment MapboxAdminConfig_mapToken on MapToken {
  id
  publicToken
  secretToken
  provider
  styleOwner
  styleId
  styles {
    id
    owner
    name
    previewUrl
    createdAt
    updatedAt
    isCurrent
  }
}
",
            },
          },
        }
      }
      render={[Function]}
      variables={
        Object {
          "provider": "MAPBOX",
        }
      }
    />
  </div>
</div>
`;
